agent/collector.go

```diff
@@ -270,12 +271,23 @@ func (l *Collector) postToAPI(endpoint config.APIEndpoint, checkPath string, bod
 	req.Header.Add("X-Dd-Hostname", l.cfg.HostName)
 	req.Header.Add("X-Dd-Processagentversion", Version)
 
+	// we create a context with timeout and attach it to the http request, to make sure that it doesn't get stuck on
+	// any circumstances
```

Comment:
This comment seems unnecessary to me as this is implied by having a context timeout.